implicitly importing a package , iverilog cannot find declarations in package.

pkt as follows:

```
`timescale 1ps/1ps
package test_pkt;
    parameter A = 3'd0;
    parameter B = 3'd2;
    function [3:0] add;
        input [3:0] a;
        input [3:0] b;

        add = a+b;
    endfunction
endpackage
```

test.sv as follows:
```
import test_pkt::*;

`timescale 1ps/1ps
module test;

reg[3:0] to;

initial begin
    to = add(A,B);
    # 100ns
    $finish();
end
    
endmodule
```
compile with:

`iverilog -g2012 *.sv`

ERROR results:
```
test.sv:9: error: No function named `add' found in this context (test).
1 error(s) during elaboration.
```

--- a/ivtest/ivltests/sv_wildcard_import8.v
+++ b/ivtest/ivltests/sv_wildcard_import8.v
@@ -1,0 +1,28 @@
+// Check that implicit imports of functions and tasks works if the wildcard
+// import statement is in the unit scope.
+
+package P;
+
+  function integer f(integer x);
+    return x * 2;
+  endfunction
+
+  task t(bit failed);
+    if (failed) begin
+      $display("FAILED");
+    end else begin
+      $display("PASSED");
+    end
+  endtask
+
+endpackage
+
+import P::*;
+
+module test;
+
+  initial begin
+    t(f(10) !== 20);
+  end
+
+endmodule

--- a/ivtest/regress-vvp.list
+++ b/ivtest/regress-vvp.list
@@ -41,6 +41,7 @@ sv_array_cassign6		vvp_tests/sv_array_cassign6.json
 sv_array_cassign7		vvp_tests/sv_array_cassign7.json
 sv_foreach9			vvp_tests/sv_foreach9.json
 sv_foreach10			vvp_tests/sv_foreach10.json
+sv_wildcard_import8		vvp_tests/sv_wildcard_import8.json
 sdf_header			vvp_tests/sdf_header.json
 task_return1			vvp_tests/task_return1.json
 task_return2			vvp_tests/task_return2.json

--- a/ivtest/vvp_tests/sv_wildcard_import8.json
+++ b/ivtest/vvp_tests/sv_wildcard_import8.json
@@ -1,0 +1,5 @@
+{
+    "type"          : "normal",
+    "source"        : "sv_wildcard_import8.v",
+    "iverilog-args" : [ "-g2005-sv" ]
+}

--- a/main.cc
+++ b/main.cc
@@ -1126,6 +1126,8 @@ int main(int argc, char*argv[])
 	    rc += pform_parse(source_files[idx]);
       }
 
+      pform_finish();
+
       if (pf_path) {
 	    ofstream out (pf_path);
 	    out << "PFORM DUMP NATURES:" << endl;

--- a/parse_api.h
+++ b/parse_api.h
@@ -58,6 +58,8 @@ extern void pform_dump(std::ostream&out, const PTaskFunc*tf);
  */
 extern int pform_parse(const char*path);
 
+extern void pform_finish();
+
 extern std::string vl_file;
 
 extern void pform_set_timescale(int units, int prec, const char*file,

--- a/pform.cc
+++ b/pform.cc
@@ -405,17 +405,22 @@ LexicalScope* pform_peek_scope(void)
       return lexical_scope;
 }
 
-void pform_pop_scope()
+static void pform_check_possible_imports(LexicalScope *scope)
 {
-      LexicalScope*scope = lexical_scope;
-      assert(scope);
-
       map<perm_string,PPackage*>::const_iterator cur;
       for (cur = scope->possible_imports.begin(); cur != scope->possible_imports.end(); ++cur) {
             if (scope->local_symbols.find(cur->first) == scope->local_symbols.end())
                   scope->explicit_imports[cur->first] = cur->second;
       }
       scope->possible_imports.clear();
+}
+
+void pform_pop_scope()
+{
+      LexicalScope*scope = lexical_scope;
+      assert(scope);
+
+      pform_check_possible_imports(scope);
 
       lexical_scope = scope->parent_scope();
       assert(lexical_scope);
@@ -3405,4 +3410,12 @@ int pform_parse(const char*path)
 
       destroy_lexor();
       return error_count;
+}
+
+void pform_finish()
+{
+      // Wait until all parsing is done and all symbols in the unit scope are
+      // known before importing possible imports.
+      for (auto unit : pform_units)
+	    pform_check_possible_imports(unit);
 }

